### DISCUSSION TITLE
Tx partial signed with @solana/web3.js cant be partial signed in solnet.

### DISCUSSION THREAD

**Discussion by timaiv** [bmresearch/Solnet#396] [Q&A]
Web3.js(1.44.0) byte array after serializeMessage() is not equal to compileMessage() in solnet(5.0.6).
Account order is different. Debugging further.

```js
let tx = Transaction.from(data)
let compiled = tx.serializeMessage();
```
```cs
var txDeserialized = Solnet.Rpc.Models.Transaction.Deserialize(data);
var compiled = txDeserialized.CompileMessage();
```
Same operation success in web3.js, tx sent successfully.

**Comment by timaiv** [comment]
This caused by `x.pubkey.toBase58().localeCompare(y.pubkey.toBase58());` additional check. [Link](https://github.com/solana-labs/solana/blob/29b597cea5410805e559c200119746330fbfdd46/web3.js/src/transaction.ts#L412)
```js    
// Sort. Prioritizing first by signer, then by writable
uniqueMetas.sort(function (x, y) {
      if (x.isSigner !== y.isSigner) {
        // Signers always come before non-signers
        return x.isSigner ? -1 : 1;
      }
      if (x.isWritable !== y.isWritable) {
        // Writable accounts always come before read-only accounts
        return x.isWritable ? -1 : 1;
      }
      // Otherwise, sort by pubkey, stringwise.
      return x.pubkey.toBase58().localeCompare(y.pubkey.toBase58());
    });
```

**Comment by tiago18c** [comment]
I think this should be easily fixed.

The problem was web3.js had issues with partial signing and order of keys changing. We were kinda compatible, but there were edge cases. Looking at git blame, that was changed this past week to have deterministic order of keys!

Will create an issue and work on this during the weekend!

**Reply by timaiv** [reply]
Now bytes are same, added sort by pubkey as in web3.js
```cs
        internal IList<AccountMeta> AccountList
        {
            get
            {
                List<AccountMeta> res = _accounts.Select(acc => acc).ToList();

                res.Sort((x, y) =>
                {
                    if (x.IsSigner != y.IsSigner)
                    {
                        // Signers always come before non-signers
                        return x.IsSigner ? -1 : 1;
                    }
                    if (x.IsWritable != y.IsWritable)
                    {
                        // Writable accounts always come before read-only accounts
                        return x.IsWritable ? -1 : 1;
                    }
                    // Otherwise, sort by pubkey, stringwise.
                    return x.PublicKey.CompareTo(y.PublicKey);
                });

                return res;
            }
        }
```

**Reply by hoakbuilds** [reply]
@timaiv do you mind opening a PR for this? there's an associated issue at #397, if not then we'll open up a PR later

**Reply by timaiv** [reply]
@murlokito Sorry, no,  Also I would be thankful for adding "PartialSign" method, that finds corresponding account in signatures.
